kong/developer_role_service_test.go

```diff
@@ -50,15 +56,25 @@ func TestDeveloperRoleService(T *testing.T) {
 
 	err = client.DeveloperRoles.Delete(defaultCtx, createdRole.ID)
 	assert.Nil(err)
+
+	defaultWorkspace.Config = map[string]interface{}{"portal": false}
+	_, err = client.Workspaces.Update(defaultCtx, defaultWorkspace)
```

Comment:
Working through this, basic helpers to just enable and disable the Portal seemed incomplete, since other tests will need to set additional workspace settings beyond that (e.g. the developer auth stuff in #27) and should clear out _all_ changed configuration once they complete.

To accommodate that, my initial prototype had `setupPortal()` take a `map[string]interface{}` of additional workspace settings and applied those along with `{"portal": true}`. `disablePortal()` didn't have any good way to revert the additional settings though, so you'd be stuck doing that by hand.

a733496 should be a more generic solution that reduces boilerplate checks for client workspace request success and provides a means to clean up any changes you made after. Tests that only need to enable the Portal and not any related settings do need to pass `{"portal": true}` explicitly, but that seems reasonable enough.

Reply:
It seems quite good to me. As it covers all the actual use cases you can stick to the `map[string]interface{}`. Create a simple map is quite reasonable indeed.